### PR TITLE
add eventLoggers on batch changes

### PR DESCRIPTION
**Pull request by malomarrec**
Adding metrics for our new analytics pipeline.
I am tentatively using a new metrics format to improve maintainability and legibility: `context:object:action`.

Follow-up work to add:

batch_change_preview:expand_changeset:clicked
batch_change_preview:action_publish_on_apply:clicked
batch_change_preview:action_publish_dradf_on_apply:clicked
batch_change_preview:action_unpublish_on_apply:clicked

batch_change_details:burndow_chart_tab:clicked
batch_change_details:executions_tab:clicked
batch_change_details:archived_tab:clicked
batch_change_details:bulk_operations_tab:clicked
batch_change_details:user_breadcrumb:clicked

TODO: change tutorials - remove exlusionary terms tuto

## Test plan

Metrics

## App preview:

- [Web](https://sg-web-mm-add-metrics.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ldwtxsdedb.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.tsx [hunk 1]
```diff
@@ -524,7 +529,8 @@ const WorkspaceStep: React.FunctionComponent<React.PropsWithChildren<WorkspaceSt
                 <Card className={classNames('mt-2', styles.stepCard)}>
                     <CardBody>
                         {!step.skipped && (
-                            <Tabs className={styles.stepTabs} size="small" behavior="forceRender">
+                            // TODO: get tab name instead of index
```
Comment (eseliger) [inline client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.tsx:532]: reminder, here's a todo left 😬

**Review by eseliger — APPROVED** [review]
LGTM, there's a bunch for code formatting things but the linter/prettier will throw that at you in CI anyways, so not blocking :)

File: client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.tsx (hunk 1 above)
Comment (malomarrec) [reply client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.tsx]: I implemented something, but it requires a hardcoded list of tabs name (not great). Thoughts?

Comment (malomarrec) [reply client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.tsx]: @courier-new if you have feedback on this implementation, I'd love to hear it: https://github.com/sourcegraph/sourcegraph/pull/39207/commits/6c229081fd91895d9adf5dec08308b0e9c96d207

Comment (courier-new) [reply client/web/src/enterprise/batches/batch-spec/execute/workspaces/WorkspaceDetails.tsx]: Hey Malo, I didn't have a chance to look at this today but I will offer feedback as soon as I can. 😄

**Pull request merged**